src/arrayview.rs

```diff
@@ -267,25 +329,52 @@ mod test {
         assert_eq!(65535, x.y());
     }
 
-    #[test]
-    #[allow(dead_code)]
-    fn test_slice() {
-        define_struct!(
-            A,
-            RefA,
-            RefMutA,
-            "no_schema",
-            4,
-            (value, set_value, u32, 0, 32)
-        );
-
-        let mut v: Vector<A> = Vector::with_len(10);
-        for i in 0..10 {
+    fn create_values(size: usize) -> Vector<Value> {
+        let mut v: Vector<Value> = Vector::with_len(size);
+        for i in 0..size as u32 {
             let mut a = v.at_mut(i as usize);
             a.set_value(i);
         }
+        v
+    }
 
-        let view: ArrayView<_> = v.as_view();
+    #[test]
+    fn from() {
+        let v = create_values(10);
+        let _x: ArrayView<_> = (&v).into();
```

Comment:
Should not we test here something? Like len?

Reply:
Done